content/implement/client.md

```diff
@@ -0,0 +1,209 @@
+
+If the endpoint you are consuming requires a payload, then it will auto generate a payload
+struct that mimics the type that is defined in the design package...
+
+```go
+// RateBottlePayload is the bottle rate action payload.
+type RateBottlePayload struct {
+	Rating int `form:"rating" json:"rating" xml:"rating"`
+}
+```
+
+#### A Path Generator
+
+The path generator will take any url params that are defined in the definition and return
+the proper path that the request is supposed to go to...
+
+```go
+// RateBottlePath computes a request path to the rate action of bottle.
+func RateBottlePath(accountID int, bottleID int) string {
+	param0 := strconv.Itoa(accountID)
+	param1 := strconv.Itoa(bottleID)
+
+	return fmt.Sprintf("/cellar/accounts/%s/bottles/%s/actions/rate", param0, param1)
+}
+```
+
+#### A method that creates the request and executes it
+
+If there is no need to modify the request, then you can simply call the function that was
+made with the endpoint name...
+
+```go
+// RateBottle makes a request to the rate action endpoint of the bottle resource
+func (c *Client) RateBottle(ctx context.Context, path string, payload *RateBottlePayload, contentType string) (*http.Response, error) {
+	req, err := c.NewRateBottleRequest(ctx, path, payload, contentType)
+	if err != nil {
+		return nil, err
+	}
+	return c.Client.Do(ctx, req)
+}
+```
+
+#### A method that creates the request and returns it
+
+If there is some need to alter the request in some way before executing it, you can get
+the request and then handle executing it later...
+
+```go
+// NewRateBottleRequest create the request corresponding to the rate action endpoint of the bottle resource.
+func (c *Client) NewRateBottleRequest(ctx context.Context, path string, payload *RateBottlePayload, contentType string) (*http.Request, error) {
+	var body bytes.Buffer
+	if contentType == "" {
+		contentType = "*/*" // Use default encoder
+	}
+	err := c.Encoder.Encode(payload, &body, contentType)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode body: %s", err)
+	}
+	scheme := c.Scheme
+	if scheme == "" {
+		scheme = "http"
+	}
+	u := url.URL{Host: c.Host, Scheme: scheme, Path: path}
+	req, err := http.NewRequest("PUT", u.String(), &body)
+	if err != nil {
+		return nil, err
+	}
+	header := req.Header
+	if contentType == "*/*" {
+		header.Set("Content-Type", "application/json")
+	} else {
+		header.Set("Content-Type", contentType)
+	}
+	return req, nil
+}
+```
+
+#### A method which decodes the http response to a goa type
+
+If there is data returned in the form of a `MediaType` then goa will also generate code to
+take the http response and convert it back to a native data type...
+
+```go
+// DecodeAccount decodes the Account instance encoded in resp body.
+func (c *Client) DecodeAccount(resp *http.Response) (*Account, error) {
+	var decoded Account
+	err := c.Decoder.Decode(&decoded, resp.Body, resp.Header.Get("Content-Type"))
+	return &decoded, err
+}
+```
+
+### Example:
```

Comment:
Would be nice to have a section "Putting it all together" that shows how to use the previously described components to put together a request. The example is great, I would just add a bit more description and context around it.